specification/vmware/Microsoft.AVS/routes.tsp

```diff
@@ -0,0 +1,1362 @@
+import "@azure-tools/typespec-azure-core";
+import "@typespec/rest";
+import "./models.tsp";
+import "./parameters.tsp";
+
+using TypeSpec.Rest;
+using TypeSpec.Http;
+using Azure.ResourceManager;
+
+namespace Microsoft.AVS;
+
+interface Operations extends Azure.ResourceManager.Operations {}
+
+interface Locations {
+  @doc("Return trial status for subscription by region")
+  @route("/subscriptions/{subscriptionId}/providers/Microsoft.AVS/locations/{location}/checkTrialAvailability")
+  @post
+  checkTrialAvailability is Azure.Core.Foundations.Operation<
+    {
+      ...SubscriptionIdParameter;
+      ...LocationParameter;
+
+      @doc("The sku to check for trial availability")
+      @body
+      sku: Sku;
+    },
+    Trial
+  >;
+
+  @doc("Return quota for subscription by region")
+  @route("/subscriptions/{subscriptionId}/providers/Microsoft.AVS/locations/{location}/checkQuotaAvailability")
+  @post
+  checkQuotaAvailability is Azure.Core.Foundations.Operation<
+    {
+      ...SubscriptionIdParameter;
+      ...LocationParameter;
+    },
+    Quota
+  >;
+}
+
+@armResourceOperations
+interface PrivateClouds {
+  list is ArmResourceListByParent<PrivateCloud>;
+  listInSubscription is ArmListBySubscription<PrivateCloud>;
+  get is ArmResourceRead<PrivateCloud>;
+  createOrUpdate is ArmResourceCreateOrUpdateAsync<PrivateCloud>;
+  update is ArmResourcePatchAsync<PrivateCloud, PrivateCloudUpdateProperties>;
+  delete is ArmResourceDeleteAsync<PrivateCloud>;
+
+  @doc("Rotate the vCenter password")
+  @route("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AVS/privateClouds/{privateCloudName}/rotateVcenterPassword")
+  @post
+  rotateVcenterPassword is Azure.Core.Foundations.Operation<
+    {
+      ...SubscriptionIdParameter;
+      ...ResourceGroupParameter;
+      ...PrivateCloudNameParameter;
+    },
+    void
+  >;
+
+  @doc("Rotate the NSX-T Manager password")
+  @route("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AVS/privateClouds/{privateCloudName}/rotateNsxtPassword")
+  @post
+  rotateNsxtPassword is Azure.Core.Foundations.Operation<
+    {
+      ...SubscriptionIdParameter;
+      ...ResourceGroupParameter;
+      ...PrivateCloudNameParameter;
+    },
+    void
+  >;
+
+  @doc("List the admin credentials for the private cloud")
+  @route("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AVS/privateClouds/{privateCloudName}/listAdminCredentials")
+  @post
+  listAdminCredentials is Azure.Core.Foundations.Operation<
+    {
+      ...SubscriptionIdParameter;
+      ...ResourceGroupParameter;
+      ...PrivateCloudNameParameter;
+    },
+    AdminCredentials
+  >;
+}
+
+interface Clusters {
```

Comment:
You should just add Cluster as a child resource, by setting `@parentResource` for cluster, as described here: https://azure.github.io/typespec-azure/docs/getstarted/azure-resource-manager/step03

Then you can just use the standard operation templates.